tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -189,6 +189,154 @@ public void SetParameters(TestParameters parameters)
         }
     }
 
+    /// <summary>
+    /// Set benchmark metadata
+    /// </summary>
+    /// <param name="hostInfo">Host info</param>
+    /// <param name="jobInfo">Job info</param>
+    public void SetBenchmarkMetadata(BenchmarkHostInfo hostInfo, BenchmarkJobInfo jobInfo)
```

Comment:
Mostly thinking out loud, but I don't see any reason these all need or _should_ be in Datadog.trace? These could all be added to benchmarkdotnet, no? Or are they _intentionally_ public for people to _manually_ set them? 

Given the prevalence of Benchmarkdotnet, I'd personally be inclined not to make them public unless we have a specific ask. Hard to imagine people are benchmarking with anything else, no?

Reply:
I get it but the Idea is to have it in the public api as other languages. Also, I'm planning to add this support for `TimeItSharp` and the crank importer.

Reply:
Fair enough 👍